### ISSUE TITLE
Prelude.chr: bad argument: 4651120416 from Data.Attoparsec.Text.inClass?

### ISSUE THREAD

**Issue by elaforge** [bos/attoparsec#103]
Ever since upgrading many things, including likely attoparsec (I lost my old install, so I don't know what version I was at), I've been getting occasional mostly non-deterministic exceptions like "Prelude.chr: bad argument: 4651120416".  The number varies.  I used +RTS -xc, and it fingered a function that contains a line like:

```
import qualified Data.Attoparsec.Text as A
...
    ident <- (if null_ok then A.takeWhile else A.takeWhile1)
        (A.notInClass (until ++ " \n\t|=)"))
```

I replaced the notInClass with a \c -> c /= ... type function, and no more exceptions.  So perhaps Text.FastSet has a bug.  I get the exception with both 0.12.1.6 and 0.13.0.0.

I recompiled attoparsec with -fprof-auto-top to try to get a more specific location, but I couldn't get any exceptions to happen.  But if it's memory corruption then it's not a big surprise if all the rearrangement implied by SCC annotations causes it to disappear.

I've been thinking of a binary search on attoparsec versions to try to find where it starts happening, but before I go to the bother, any other ideas?

**Comment by spl** [comment]
The test suite failed for me:

```
Preprocessing test suite 'tests' for attoparsec-0.13.0.0...
[ 1 of 25] Compiling QC.Rechunked     ( tests/QC/Rechunked.hs, dist/build/tests/tests-tmp/QC/Rechunked.dyn_o )
[...]
[25 of 25] Compiling Main             ( tests/QC.hs, dist/build/tests/tests-tmp/Main.dyn_o )
Linking dist/build/tests/tests ...
running tests
Running 1 test suites...
Test suite tests: RUNNING...
bs:
  anyWord8: [OK, passed 100 tests]
[...]
  members: [OK, passed 100 tests]
  nonmembers: [Failed]
*** Failed! Exception: 'Prelude.chr: bad argument: 140737122496936' (after 11 tests and 1 shrink):
"xA\EMbR"
"J\SYN\ETBq"
(used seed -2009180548307453083)

         Properties   Total
 Passed  64           64
 Failed  1            1
 Total   65           65
Test suite tests: FAIL
Test suite logged to: dist/test/attoparsec-0.13.0.0-tests.log
0 of 1 test suites (0 of 1 test cases) passed.
builder for ‘/nix/store/v5gbfalxp6wvpj2f4hjmak3aq27p2cqp-haskell-attoparsec-0.13.0.0.drv’ failed with exit code 1
```

**Comment by bos** [comment]
This is the second report I've gotten recently of a package being broken under Nix, and only under Nix.

I hope you don't mind if I ask for some further evidence that the problem lies with attoparsec and not something broken in Nix.

**Comment by elaforge** [comment]
I'm not using nix.

**Comment by bos** [comment]
Cool, I'll dig in then.

**Comment by elaforge** [comment]
Thanks.  Let me know if I can help out.  It's not easy to reproduce, but in the past it would eventually happen, just maybe after a week of use.

**Comment by bos** [comment]
Fix issued as 0.13.0.1.
